onnx/shape_inference/implementation.h

```diff
@@ -12,6 +12,39 @@
 namespace ONNX_NAMESPACE {
 namespace shape_inference {
 
+struct SymbolicShape {
+  SymbolicShape(GraphProto* g) : index_(0){
+    existing_shape_set.clear();
+    checkExistingSymbolicShape(*g->mutable_input());
+    checkExistingSymbolicShape(*g->mutable_output());
+    checkExistingSymbolicShape(*g->mutable_value_info());
+  }
+  std::string createNew() {
```

Comment:
It would be useful to have the symbol_prefix as an optional parameter to this function (with a default value "unk__"). This allows callers to create more meaningful names if they want.